tests/tests/pushdown.rs

```diff
@@ -146,3 +146,165 @@ fn pushdown(mut conn: PgConnection) {
         );
     }
 }
+
+#[rstest]
+fn pushdown_is_not_null(mut conn: PgConnection) {
+    let sql = r#"
+        CREATE TABLE test (
+            id SERIAL8 NOT NULL PRIMARY KEY,
+            col_boolean boolean DEFAULT false,
+            col_text text,
+            col_int8 int8
+        );
+    "#;
+    sql.execute(&mut conn);
+
+    let sql = r#"
+        CREATE INDEX idxtest ON test USING bm25 (id, col_boolean, col_text, col_int8)
+        WITH (key_field='id', text_fields = '{"col_text": {"fast": true, "tokenizer": {"type":"raw"}}}');
+    "#;
+    sql.execute(&mut conn);
+
+    "INSERT INTO test (id, col_text) VALUES (1, NULL);".execute(&mut conn);
+    "INSERT INTO test (id, col_text) VALUES (2, 'foo');".execute(&mut conn);
+    "INSERT INTO test (id, col_text, col_int8) VALUES (3, 'bar', 333);".execute(&mut conn);
+    "INSERT INTO test (id, col_int8) VALUES (4, 444);".execute(&mut conn);
+
+    "SET enable_indexscan TO off;".execute(&mut conn);
+    "SET enable_bitmapscan TO off;".execute(&mut conn);
+    "SET max_parallel_workers TO 0;".execute(&mut conn);
+
+    let sql = r#"
+        EXPLAIN (ANALYZE, VERBOSE, FORMAT JSON)
+        SELECT count(*)
+        FROM test
+        WHERE col_text IS NOT NULL
+        AND id @@@ '1';
+    "#;
+
+    eprintln!("/----------/");
+    eprintln!("{sql}");
+
+    let (plan,) = sql.fetch_one::<(Value,)>(&mut conn);
+    eprintln!("{plan:#?}");
+
+    // Verify that the custom scan is used
+    let plan = plan
+        .pointer("/0/Plan/Plans/0")
+        .unwrap()
+        .as_object()
+        .unwrap();
+    pretty_assertions::assert_eq!(
+        plan.get("Node Type"),
+        Some(&Value::String(String::from("Custom Scan")))
+    );
+
+    // Verify that count is correct
+    let count = r#"
+        SELECT count(*)
+        FROM test
+        WHERE col_text IS NOT NULL
+        AND id @@@ paradedb.range(field=> 'id', range=> '[1, 5]'::int8range);
+    "#
+    .fetch::<(i64,)>(&mut conn);
+    assert_eq!(count, vec![(2,)]);
+
+    let count = r#"
```

Comment:
I am relatively new to the p[roject, but a little concerned with this style of unit testing where a lot of assertions are in a single executable test. Can we factor this out into multiple tests?
```rust
pushdown_is_not_null_count_is_correct
pushdown_is_not_null_multi_predicates_works
pushdown_is_not_null_aggregations_are_correct
pushdown_is_not_null_distinct_correct
pushdown_is_not_null_join_works
```

Reply:
@edmondop this is a great suggestion, I'm breaking them into multiple tests.